lib/orocos/process.rb

```diff
@@ -1171,9 +1198,25 @@ def self.wait_running(process, timeout = nil, name_service = Orocos::CORBA.name_
                     block.call(process)
                 else
                     # Get any task name from that specific deployment, and check we
-                    # can access it. If there is none
+                    # can access it either through the ior pipe or through the CORBA name
+                    # service.
+                    if ior_read?
+                        ior_message += ior_read.read_nonblock 1
```

Comment:
'1' here is 'one byte'.

You can put 4096 without a problem. It will return a partial buffer if there is not that much data.